client/web/compose/src/lib/record-filter.js

```diff
@@ -32,7 +32,7 @@ export function getFieldFilter (name, kind, query = '', operator = '=') {
   let boolQuery = query
   let numQuery = query
 
-  if (typeof query === 'object' && query !== null) {
+  if (typeof query !== 'object' && query !== null) {
     boolQuery = toBoolean(query)
```

Comment:
Probably makes sense to move this right after line 60, since you don't need it anywhere else.